internal/pkg/manifest/workload.go

```diff
@@ -441,14 +441,78 @@ func (p *PlacementArgs) isEmpty() bool {
 // PlacementString represents what types of subnets (public or private subnets) to place tasks.
 type PlacementString string
 
+// SecurityGroupsIDsOrConfig represents security groups attached to task. It supports unmarshalling
+// yaml which can either be of type SecurityGroupsConfig or a list of strings.
+type SecurityGroupsIDsOrConfig struct {
+	SecurityGroupIds     []string `yaml:"security_groups"`
+	SecurityGroupsConfig SecurityGroupsConfig
+}
+
+func (s *SecurityGroupsIDsOrConfig) IsEmpty() bool {
+	return (s.SecurityGroupIds == nil || len(s.SecurityGroupIds) == 0) && s.SecurityGroupsConfig.isEmpty()
```

Comment:
```suggestion
	return len(s.SecurityGroupIds) == 0 && s.SecurityGroupsConfig.isEmpty()
```
since `s.SecurityGroupIds` is `[]string`. Same thing below.

Reply:
change is made in this file at all places.